src/datachain/catalog/catalog.py

```diff
@@ -1178,12 +1180,15 @@ def register_dataset(
         version: int,
         target_dataset: DatasetRecord,
         target_version: Optional[int] = None,
+        session: Optional["Session"] = None,
```

Comment:
Let's think about this. Catalog belong to session, and now we pass back session to catalog - it's like a circular dependency. Something is wrong with the design on these objects to my mind.

Can we call `session.add_dataset_version(target_dataset, target_version)` in places where we run `create_dataset`? (to avoid catalog operating on the Session)

Reply:
Thanks @shcheklein. Moved it to dataset query back. It is working as expected.